src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/tools/ToolInstallTask.groovy

```diff
@@ -72,36 +59,36 @@ class ToolInstallTask extends DefaultTask {
             def json = builder.create().toJson(tools)
             toolFile.text = json
         } else {
-            ToolJson[] tools = [tool]
+            WPIToolInfo[] tools = [tool]
             def json = builder.create().toJson(tools)
             toolFile.text = json
         }
     }
 
     private File getScriptFile() {

```

Comment:
If this is a task input or output it should be annotated with `@InputFile` or `@InputDirectory` depending upon what it is.

Reply:
The toolFile kind of is an output, but we want this to always run.

Reply:
The scriptFile is also just a private function, not an input or output. And InputFile doesn't work from grabbing from embedded resources in the plugin jar.

Reply:
If you always want it to run you need to explicitly tell gradle that it's never up to date then.
I think it's something like `outputs.upToDateWhen` which, from a cursory look, I don't think you're doing.

https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskOutputs.html#upToDateWhen-org.gradle.api.specs.Spec-

Reply:
It also always runs if you declare no outputs.

Reply:
Ah, right.